zerver/lib/actions.py

```diff
@@ -4547,6 +4549,19 @@ def try_update_realm_custom_profile_field(realm: Realm, field: CustomProfileFiel
     field.save(update_fields=['name'])
     notify_realm_custom_profile_fields(realm, 'update')
 
+def try_move_realm_custom_profile_field(realm: Realm, field: CustomProfileField,
+                                        direction: Text) -> None:
+    if direction == 'up':
+        neighbour_field = CustomProfileField.objects.filter(order__lt=field.order).order_by('order').last()
+    else:
+        neighbour_field = CustomProfileField.objects.filter(order__gt=field.order).order_by('order').first()
+    if neighbour_field is None:
+        raise JsonableError(_("Can't move the field."))
+    neighbour_field.order, field.order = field.order, neighbour_field.order
+    field.save(update_fields=['order'])
+    neighbour_field.save(update_fields=['order'])
+    notify_realm_custom_profile_fields(realm, 'update')
```

Comment:
I think a better API for this would be to send a mapping from (field_id -> order), and re-sort the fields to match that order in the backend.  That way, if we later change the frontend to support a different model (e.g. to be able to drag the fields around to re-order them), we'd just need to change the frontend to make it work.